src/Metadata/TargetsMetadata.php

```diff
@@ -57,22 +58,27 @@ protected static function getSignedCollectionOptions(): array
                 ]),
                 'roles' => new All([
                     new Type('\ArrayObject'),
-                    new Collection([
-                        'name' => [
-                            new NotBlank(),
-                            new Type(['type' => 'string']),
-                        ],
-                        'paths' => [
-                            new Type(['type' => 'array']),
-                            new All([
-                                new Type(['type' => 'string']),
+                    new TufCollection([
```

Comment:
What's the nature of these changes? It looks like much more has changed here than needs to, but I also don't know this code as well as you.

Reply:
Basically `Symfony\Component\Validator\Constraints\Collection` does support validating a key does **not** exist. So previously we created `\Tuf\Constraints\Collection` to support this an other future add-ons we might want. 

Currently in this code we were using `Symfony\Component\Validator\Constraints\Collection` here because in this part we didn't need exclude and field but now we do so we switch to our extended version.

This allows use to add `'unsupportedFields' => ['path_hash_prefixes'],` and it will validate that for use.

The other change is we now have to nest the previous arguments to the `Collection` constructor under `fields`. This the default option so if don't need to send more than 1 option this is assumed. But now that we want to add another option, `unsupportedFields` we need to nest. This just how default options work in Symfony Validator Constraints.

Reply:
> So previously we created \Tuf\Constraints\Collection to support this an other future add-ons we might want.

Which PR did we do that in?

Reply:
https://github.com/php-tuf/php-tuf/pull/99 (via git history)